### ISSUE TITLE
[Bug]: Inject crashes

### ISSUE THREAD

**Issue by APdaofen** [Yimura/YimMenu#2753]
### Describe the bug

Using both the normal and nightly versions crashed when injected
Below is the log from the nightly version

### Steps To Reproduce

Normal injection

### Expected Behavior

Hopefully this crash can be fixed

### Logs and/or screenshots

[cout.log](https://github.com/YimMenu/YimMenu/files/14371554/cout.log)

### Store

Steam

### Language

Simplified Chinese

### YimMenu Language

Chinese (CN)

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by Yimura** [comment]
Thanks for including the log file, we'll take a look whenever we get the time.

**Comment by gir489returns** [comment]
According to IDA, this is crashing at the same place in ImGui from @sch-lda's report, when it attempts to dereference rcx, but rcx is zeroed out.

Is it possible the nightly didn't scoop up your changes from #2738?

**Comment by Yimura** [comment]
> According to IDA, this is crashing at the same place in ImGui from @sch-lda's report, when it attempts to dereference rcx, but rcx is zeroed out.
> 
> Is it possible the nightly didn't scoop up your changes from #2738?

I talked to sch-ida on Discord and some of these users only have integrated graphics and 8GB RAM, go figure it barely manages to run the game. The moment a larger font is loaded in the game crashes. The default font of 20px loads perfectly fine but one of the bigger fonts doesn't.

```cpp
ImFontConfig font_msyh_cfg{};
		font_msyh_cfg.FontDataOwnedByAtlas = false;
		strcpy(font_msyh_cfg.Name, "msyh");
		g.window.font_title = io.Fonts->AddFontFromFileTTF(font_file_path.get_path().string().c_str(), 20.0f, &font_msyh_cfg, ImGui::GetIO().Fonts->GetGlyphRangesChineseFull());

		g.window.font_sub_title = g.window.font_title;
		g.window.font_small = g.window.font_title;
```
This is a snippet of the old renderer with the fix applied.

**Comment by gir489returns** [comment]
> I talked to sch-ida on Discord and some of these users only have integrated graphics and 8GB RAM, go figure it barely manages to run the game. The moment a larger font is loaded in the game crashes. The default font of 20px loads perfectly fine but one of the bigger fonts doesn't.

This sounds like a regression issue then, because they were working before right? Instead of using GetGlyphRangesChineseFull maybe we should go back to using GetGlyphRangesChineseSimplifiedCommon?

**Comment by Yimura** [comment]
> This sounds like a regression issue then, because they were working before right? Instead of using GetGlyphRangesChineseFull maybe we should go back to using GetGlyphRangesChineseSimplifiedCommon?

I talked to him about that and apparently the ranges from ImGui somehow miss quite a number of symbols from Chinese Simplified. This makes it hard to properly maintain the translations, though @sch-lda is better able to confirm this.

**Comment by gir489returns** [comment]
> > This sounds like a regression issue then, because they were working before right? Instead of using GetGlyphRangesChineseFull maybe we should go back to using GetGlyphRangesChineseSimplifiedCommon?
> 
> I talked to him about that and apparently the ranges from ImGui somehow miss quite a number of symbols from Chinese Simplified. This makes it hard to properly maintain the translations, though @sch-lda is better able to confirm this.

So we're stuck between a rock and a hard place. Obviously we want to try and capture as much of the Chinese cheating market as we can, as they are prime for paycheats, and we want to piss in their cornflakes as much as humanly possible, but at the same time, we also want it to be scaleable. There has to be a middle ground...

Maybe if we detect the user's machine has too little RAM we have to alert them it's on ChineseSimplifiedCommon and that some things may not work properly?

**Comment by Yimura** [comment]
Yeah the people having issues appear to have barely any VRAM (2GB) or don't have any actual real VRAM (shared with CPU RAM).

Perhaps we can look at where the characters used lie in the glyph ranges and see if the ImGui Glyph ranges are just wrong.
https://stackoverflow.com/a/1366113

**Comment by Yimura** [comment]
Should've been fixed with #2756
